File: Assignments/Weeklies/EnfeiZhang_GrittyP3/Question5/PersonRecord.java

import java.util.Calendar;

public class PersonRecord implements PersonalInfo{

    private Calendar creationDate;
    private int currentAge;

    @Override
    public Calendar getFileCreationDate() {
        creationDate = Calendar.getInstance();
        return creationDate;
    }

    @Override
    public int getCurrentAge(int birthYear) {
        currentAge = Calendar.getInstance ().get (Calendar.YEAR) - birthYear;
        return currentAge;
    }
}

File: Assignments/Weeklies/EnfeiZhang_GrittyP3/Question5/PersonRecord.kt
/*Enfei Zhang
  March 18, 2018
  Ms. Krasteva
  This is the a class that implements the PersonalInfo interface
 */

import java.util.Calendar

class PersonRecord : PersonalInfo {

    private var creationDate: Calendar? = null
    private var currentAge: Int = 0

    override val fileCreationDate: Calendar?
        get() {
            creationDate = Calendar.getInstance()
            return creationDate
        }

    override fun getCurrentAge(birthYear: Int): Int {
        currentAge = Calendar.getInstance().get(Calendar.YEAR) - birthYear
        return currentAge
    }
}
